spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/couchbase/CouchbaseAutoConfigurationIntegrationTests.java

```diff
@@ -45,9 +46,9 @@
 	private static final String BUCKET_NAME = "cbbucket";
 
 	@Container
-	static final CouchbaseContainer couchbase = new CouchbaseContainer().withClusterAdmin("spring", "password")
+	static final CouchbaseContainer couchbase = new CouchbaseContainer().withCredentials("spring", "password")
 			.withStartupAttempts(5).withStartupTimeout(Duration.ofMinutes(10))
-			.withNewBucket(BucketSettings.create(BUCKET_NAME));
+			.withBucket(new BucketDefinition(BUCKET_NAME));
 
 	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
```

Comment:
@snicoll FYI `"spring.couchbase.connection-string:localhost:" + couchbase.getMappedPort(11210)` isn't entirely correct.

Consider using `"spring.couchbase.connection-string:" + couchbase.getConnectionString()` (takes `getContainerIpAddress()` into consideration).

And, if more services are tested, the environment needs to be configured:
https://www.testcontainers.org/modules/databases/couchbase/#usage-example

Reply:
Ah that's good to know, I've polished that in https://github.com/spring-projects/spring-boot/commit/04544f98428cae4c3083f0ee58410eea3591f066